src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/FeaturizedReadSets.java

```diff
@@ -123,15 +125,74 @@ private static List<Integer> featurize(final GATKRead read, final VariantContext
+            byte[] haplotypeBases = haplotype.getBases();
+            final SmithWatermanAlignment readToHaplotypeAlignment = aligner.align(haplotypeBases, read.getBases(), SmithWatermanAlignmentConstants.ALIGNMENT_TO_BEST_HAPLOTYPE_SW_PARAMETERS, SWOverhangStrategy.SOFTCLIP);
             final GATKRead copy = read.copy();
             copy.setCigar(readToHaplotypeAlignment.getCigar());
-            final int mismatchCount = AlignmentUtils.getMismatchCount(copy, haplotype.getBases(), readToHaplotypeAlignment.getAlignmentOffset()).numMismatches;
+            final int mismatchCount = AlignmentUtils.getMismatchCount(copy, haplotypeBases, readToHaplotypeAlignment.getAlignmentOffset()).numMismatches;
             result.add(mismatchCount);
 
             final long indelsVsBestHaplotype = readToHaplotypeAlignment.getCigar().getCigarElements().stream().filter(el -> el.getOperator().isIndel()).count();
             result.add((int) indelsVsBestHaplotype);
+
+            final int readStartInHaplotype = readToHaplotypeAlignment.getAlignmentOffset();
+            final AlignmentStateMachine asm = new AlignmentStateMachine(copy);
+            asm.stepForwardOnGenome();
+            final int[] featuresWithin5Bases = new int[5];
+            final int[] featuresWithin10Bases = new int[5];
+            final int[] featuresWithin25Bases = new int[5];
+            final int[] featuresWithin50Bases = new int[5];
+            while (!asm.isRightEdge()) {
+                final PileupElement pe = asm.makePileupElement();
+                final int distanceFromVariant = Math.abs(asm.getReadOffset() - readPositionOfVariantStart);
+
+                int[] featuresToAddTo;
+                if (distanceFromVariant <= 5) {
+                    featuresToAddTo = featuresWithin5Bases;
+                } else if (distanceFromVariant <= 10) {
+                    featuresToAddTo = featuresWithin10Bases;
+                } else if (distanceFromVariant <= 25) {
+                    featuresToAddTo = featuresWithin25Bases;
+                } else if (distanceFromVariant <= 50) {
+                    featuresToAddTo = featuresWithin50Bases;
+                } else {
+                    asm.stepForwardOnGenome();
+                    continue;
+                }
+
+                if (pe.isAfterInsertion()) {
+                    featuresToAddTo[0]++;
+                }
+
+                if (pe.isDeletion()) {
+                    featuresToAddTo[1]++;
+                } else {
+                    final byte base = pe.getBase();
+                    final byte qual = pe.getQual();
+                    final byte haplotypeBase = haplotypeBases[asm.getGenomeOffset() + readStartInHaplotype];
+
+                    if (base != haplotypeBase) {
+                        featuresToAddTo[2]++;
+                    }
+
+                    if (qual < 10) {
```

Comment:
Nit:  magic constants here as well.  
I make it a nit here because it is rather obvious.

Reply:
fixed